pkg/virt-controller/watch/vm.go

```diff
@@ -179,14 +177,22 @@ func NewVMController(vmiInformer cache.SharedIndexInformer,
 	return c, nil
 }
 
-type sarProxy struct {
+type authProxy struct {
 	client kubecli.KubevirtClient
 }
 
-func (p *sarProxy) Create(sar *authv1.SubjectAccessReview) (*authv1.SubjectAccessReview, error) {
+func (p *authProxy) CreateSar(sar *authv1.SubjectAccessReview) (*authv1.SubjectAccessReview, error) {
 	return p.client.AuthorizationV1().SubjectAccessReviews().Create(context.Background(), sar, v1.CreateOptions{})
 }
 
+func (p *authProxy) GetNamespace(name string) (*k8score.Namespace, error) {
+	return p.client.CoreV1().Namespaces().Get(context.TODO(), name, v1.GetOptions{})
+}
+
+func (p *authProxy) GetDataSource(namespace, name string) (*cdiv1.DataSource, error) {
+	return p.client.CdiClient().CdiV1beta1().DataSources(namespace).Get(context.TODO(), name, v1.GetOptions{})
```

Comment:
should use informers for namespace and datasource

Reply:
done